projects/plugins/boost/app/modules/optimizations/lcp/class-lcp-optimize-img-tag.php

```diff
@@ -0,0 +1,236 @@
+<?php
+
+namespace Automattic\Jetpack_Boost\Modules\Optimizations\Lcp;
+
+use Automattic\Jetpack\Image_CDN\Image_CDN_Core;
+use WP_HTML_Tag_Processor;
+
+class LCP_Optimize_Img_Tag {
+
+	/**
+	 * LCP data.
+	 *
+	 * @var array
+	 */
+	private $lcp_data;
+
+	public function __construct( $lcp_data ) {
+		$this->lcp_data = $lcp_data;
+	}
+
+	/**
+	 * Optimize a viewport's LCP HTML.
+	 *
+	 * @param string $buffer The buffer/html to optimize.
+	 * @return string The optimized buffer, or the original buffer if no optimization was needed
+	 *
+	 * @since 4.0.0
+	 */
+	public function optimize_buffer( $buffer ) {
+		$optimization_util = new LCP_Optimization_Util( $this->lcp_data );
+		if ( ! $optimization_util->can_optimize() ) {
+			return $buffer;
+		}
+
+		// Defensive check to ensure the LCP HTML is not empty.
+		if ( empty( $this->lcp_data['html'] ) ) {
+			return $buffer;
+		}
+
+		// Only optimize if the type is one we know how to handle.
+		if ( $this->lcp_data['type'] !== LCP::TYPE_IMAGE ) {
+			return $buffer;
+		}
+
+		/*
+		 * Quickly check if the tag is in the buffer and return early if it's not found.
+		 * The HTML returned from cloud may not have a forward slash at the end of the tag, even if the original HTML had one.
+		 * By removing the last character from the LCP HTML, we can quickly check if the tag is in the buffer.
+		 *
+		 * `substr( '<img src="...">', 0, -1 )` -> `<img src="..."`
+		 */
+		if ( ! str_contains( $buffer, substr( $this->lcp_data['html'], 0, -1 ) ) ) {
+			return $buffer;
+		}
```

Comment:
I don't quite understand this part. If the img we get uses a forward slash (`<img src="..."/>`), how does removing the last character help us? We're still left with the forward slash at the end.